app/src/test/java/org/simple/clinic/summary/linkId/LinkIdWithPatientSheetControllerTest.kt

```diff
@@ -0,0 +1,72 @@
+package org.simple.clinic.summary.linkId
+
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.never
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.whenever
+import io.reactivex.Single
+import io.reactivex.subjects.PublishSubject
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.simple.clinic.patient.PatientMocker
+import org.simple.clinic.patient.PatientRepository
+import org.simple.clinic.patient.businessid.Identifier
+import org.simple.clinic.util.RxErrorsRule
+import org.simple.clinic.widgets.UiEvent
+import java.util.UUID
+
+class LinkIdWithPatientSheetControllerTest {
+
+  @get:Rule
+  val rxErrorsRule = RxErrorsRule()
+
+  private val patientRepository = mock<PatientRepository>()
+
+  private val sheet = mock<LinkIdWithPatientSheet>()
+
+  private val uiEvents = PublishSubject.create<UiEvent>()
+
+  private lateinit var controller: LinkIdWithPatientSheetController
```

Comment:
Instead of using `lateinit var`, we can use `val` here. JUnit recreates the entire class along with the properties after every test.
So instead let's do this?
`private val controller = LinkIdWithPatientSheetController(patientRepository )`

Reply:
The reason for initializing `LinkIdWithPatientSheetController` in every test is so that when we run test on the controller, it creates a clean instance every time. Here it may not make a difference right now but if for instance we start using a `clock`, we would want it to be reset with every test.

Reply:
You're right, we need clean instance every time. That's precisely what JUnit does, it recreates the entire class for every single test. So we can actually just use `val` and it will work.

Reply:
Ok. I did not know this. I will make the change.

Reply:
Done this change.